File: Groupe1/TP1/src/calculs.c
#include <stdio.h>

int main() {
    int num1, num2;
    char op;
    int result;

    printf("Entrez le premier nombre : ");
    scanf("%d", &num1);
    
    printf("Entrez le deuxième nombre : ");
    scanf("%d", &num2);
    
    printf("Entrez un opérateur (+, -, *, /, %%, &, |, ~) : ");
    scanf(" %c", &op);

    switch(op) {
        case '+':
            result = num1 + num2;
            printf("%d + %d = %d\n", num1, num2, result);
            break;
        case '-':
            result = num1 - num2;
            printf("%d - %d = %d\n", num1, num2, result);
            break;
        case '*':
            result = num1 * num2;
            printf("%d * %d = %d\n", num1, num2, result);
            break;
        case '/':
            if (num2 != 0) {
                result = num1 / num2;
                printf("%d / %d = %d\n", num1, num2, result);
            } else {
                printf("Erreur : Division par zéro.\n");
            }
            break;
        case '%':
            if (num2 != 0) {
                result = num1 % num2;
                printf("%d %% %d = %d\n", num1, num2, result);
            } else {
                printf("Erreur : Modulo par zéro.\n");
            }
            break;
        case '&':
            result = num1 & num2;
            printf("%d & %d = %d\n", num1, num2, result);
            break;
        case '|':
            result = num1 | num2;
            printf("%d | %d = %d\n", num1, num2, result);
            break;
        case '~':
            result = ~num1;
            printf("~%d = %d\n", num1, result);
            break;
        default:
            printf("Opérateur non reconnu.\n");
            break;
    }

    return 0;
}

File: Groupe1/TP1/src/calculs.py
num1 = int(input("Entrez le premier nombre : "))
num2 = int(input("Entrez le deuxième nombre : "))
op = input("Entrez un opérateur (+, -, *, /, %, &, |, ~) : ")

match op:
    case '+':
        result = num1 + num2
        print(f"{num1} + {num2} = {result}")
    case '-':
        result = num1 - num2
        print(f"{num1} - {num2} = {result}")
    case '*':
        result = num1 * num2
        print(f"{num1} * {num2} = {result}")
    case '/':
        if num2 != 0:
            result = num1 / num2
            print(f"{num1} / {num2} = {result}")
        else:
            print("Erreur : Division par zéro.")
    case '%':
        if num2 != 0:
            result = num1 % num2
            print(f"{num1} % {num2} = {result}")
        else:
            print("Erreur : Modulo par zéro.")
    case '&':
        result = num1 & num2
        print(f"{num1} & {num2} = {result}")
    case '|':
        result = num1 | num2
        print(f"{num1} | {num2} = {result}")
    case '~':
        result = ~num1
        print(f"~{num1} = {result}")
    case _:
        print("Opérateur non reconnu.")
